src/controller/CommissioneeDeviceProxy.cpp

```diff
@@ -132,6 +132,13 @@ CHIP_ERROR CommissioneeDeviceProxy::UpdateDeviceData(const Transport::PeerAddres
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR CommissioneeDeviceProxy::SetConnected()
+{
+    mState = ConnectionState::SecureConnected;
```

Comment:
Is it an CHIP_ERROR_INVALID_STATE if it is already set to SecureConnected or it is otherwise not the correct initial state?

Reply:
Also wondering - what if the loading of parameters fails - is it still ok for mState to have been changed to Connected?
or should we have this set only if parameter load works.

Reply:
added checks for current state and return of the load call.